File: Unknownbank/app/src/main/java/com/kasatria/kasatriaunknownbank/CreditCardJobDetailsScreen.kt
package com.kasatria.kasatriaunknownbank

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.saveable.rememberSaveable
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

import com.kasatria.kasatriaunknownbank.ui.theme.PrimaryBlue
import com.kasatria.kasatriaunknownbank.ui.theme.TextPrimary
import com.kasatria.kasatriaunknownbank.ui.theme.White


@Composable
fun CreditCardJobDetailsScreen(
    productName: String,
    applicationData: CreditCardApplicationData,
    onApplicationDataChange: (CreditCardApplicationData) -> Unit,
    onBack: () -> Unit,
    onNext: () -> Unit
) {




    Box(
        modifier = Modifier.fillMaxSize()
    ) {

        /*
         * Decorative background
         */

        Image(
            painter = painterResource(
                id = R.drawable.full_background_light
            ),
            contentDescription = null,
            modifier = Modifier.fillMaxSize(),
            contentScale = ContentScale.Crop
        )


        Column(
            modifier = Modifier
                .fillMaxSize()
                .statusBarsPadding()
        ) {

            /*
             * =================================
             * HEADER
             * =================================
             */

            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(56.dp)
            ) {

                IconButton(
                    onClick = onBack,
                    modifier = Modifier
                        .padding(start = 8.dp)
                        .align(Alignment.CenterStart)
                ) {

                    Icon(
                        imageVector =
                            Icons.AutoMirrored.Filled.ArrowBack,

                        contentDescription = "Back",

                        tint = TextPrimary
                    )
                }


                Text(
                    text =
                        "Applying for $productName",

                    modifier = Modifier
                        .align(Alignment.Center)
                        .padding(
                            start = 55.dp,
                            end = 30.dp
                        ),

                    color = Color(0xFF333333),

                    fontSize = 13.sp,

                    fontWeight =
                        FontWeight.Bold,

                    maxLines = 1
                )
            }


            /*
             * =================================
             * PROGRESS
             * =================================
             */

            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(
                        start = 20.dp,
                        end = 23.dp,
                        top = 20.dp
                    ),

                horizontalArrangement =
                    Arrangement.SpaceBetween,

                verticalAlignment =
                    Alignment.CenterVertically
            ) {

                Text(
                    text =
                        "Tell Us About Your Job",

                    color =
                        Color(0xFF333333),

                    fontSize = 15.sp,

                    fontWeight =
                        FontWeight.Bold
                )


                Text(
                    text = "60%",

                    color =
                        Color(0xFF0EAF5F),

                    fontSize = 14.sp,

                    fontWeight =
                        FontWeight.Medium
                )
            }


            Spacer(
                modifier =
                    Modifier.height(11.dp)
            )


            Box(
                modifier = Modifier
                    .padding(horizontal = 20.dp)
                    .fillMaxWidth()
                    .height(6.dp)
                    .background(
                        color =
                            Color(0xFFE6E8EA),

                        shape =
                            RoundedCornerShape(100.dp)
                    )
            ) {

                Box(
                    modifier = Modifier
                        .fillMaxWidth(0.60f)
                        .height(6.dp)
                        .background(
                            color =
                                Color(0xFF0EAF5F),

                            shape =
                                RoundedCornerShape(100.dp)
                        )
                )
            }


            Spacer(
                modifier =
                    Modifier.height(23.dp)
            )


            /*
             * =================================
             * FORM
             * =================================
             */

            Column(
                modifier = Modifier
                    .weight(1f)
                    .verticalScroll(
                        rememberScrollState()
                    )
                    .padding(
                        start = 20.dp,
                        end = 20.dp,
                        bottom = 100.dp
                    ),

                verticalArrangement =
                    Arrangement.spacedBy(18.dp)
            ) {

                JobTextField(
                    label = "Employer Name",
                    value = applicationData.employerName,
                    onValueChange = {
                        onApplicationDataChange(
                            applicationData.copy(
                                employerName = it
                            )
                        )
                    }
                )


                JobSelectionField(
                    label = "Occupation",
                    value = applicationData.occupation,

                    options = listOf(
                        "Sales Manager",
                        "Manager",
                        "Executive",
                        "Engineer",
                        "Consultant",
                        "Others"
                    ),

                    onSelected = {
                        onApplicationDataChange(
                            applicationData.copy(
                                occupation = it
                            )
                        )
                    }
                )


                JobSelectionField(
                    label = "Sector",
                    value = applicationData.sector,

                    options = listOf(
                        "Logistic",
                        "Banking",
                        "Technology",
                        "Retail",
                        "Manufacturing",
                        "Government",
                        "Others"
                    ),

                    onSelected = {
                        onApplicationDataChange(
                            applicationData.copy(
                                sector = it
                            )
                        )
                    }
                )


                JobSelectionField(
                    label = "Employment Type",
                    value = applicationData.employmentType,

                    options = listOf(
                        "Private Employed",
                        "Government",
                        "Self Employed"
                    ),

                    onSelected = {
                        onApplicationDataChange(
                            applicationData.copy(
                                employmentType = it
                            )
                        )
                    }
                )


                JobSelectionField(
                    label = "Business Classification",
                    value = applicationData.businessClassification,

                    options = listOf(
                        "Private Limited",
                        "Public Limited",
                        "Partnership",
                        "Sole Proprietor",
                        "Government"
                    ),

                    onSelected = {
                        onApplicationDataChange(
                            applicationData.copy(
                                businessClassification = it
                            )
                        )
                    }
                )


                JobSelectionField(
                    label = "Length of Service (Year)",
                    value = applicationData.yearsOfService,

                    options =
                        (0..30).map {
                            it.toString()
                        },

                    onSelected = {
                        onApplicationDataChange(
                            applicationData.copy(
                                yearsOfService = it
                            )
                        )
                    }
                )


                JobSelectionField(
                    label = "Length of Service (Months)",
                    value = applicationData.monthsOfService,

                    options =
                        (0..11).map {
                            it.toString()
                        },

                    onSelected = {
                        onApplicationDataChange(
                            applicationData.copy(
                                monthsOfService = it
                            )
                        )
                    }
                )
            }
        }


        /*
         * =================================
         * NEXT
         * =================================
         */

        Button(
            onClick = onNext,

            modifier = Modifier
                .align(Alignment.BottomCenter)
                .navigationBarsPadding()
                .padding(
                    start = 16.dp,
                    end = 16.dp,
                    bottom = 16.dp
                )
                .fillMaxWidth()
                .height(46.dp),

            shape =
                RoundedCornerShape(100.dp),

            colors =
                ButtonDefaults.buttonColors(
                    containerColor =
                        PrimaryBlue,

                    contentColor =
                        White
                )
        ) {

            Text(
                text = "Next",

                fontSize = 15.sp,

                fontWeight =
                    FontWeight.Bold
            )
        }
    }
}


@Composable
private fun JobTextField(
    label: String,
    value: String,
    onValueChange: (String) -> Unit
) {

    Column {

        Text(
            text = label,
            color = Color(0xFF333333),
            fontSize = 14.sp
        )


        Spacer(
            modifier =
                Modifier.height(7.dp)
        )


        BasicFormField(
            value = value,

            onValueChange =
                onValueChange,

            modifier =
                Modifier.fillMaxWidth()
        )
    }
}


@Composable
private fun JobSelectionField(
    label: String,
    value: String,
    options: List<String>,
    onSelected: (String) -> Unit
) {

    Column {

        Text(
            text = label,
            color = Color(0xFF333333),
            fontSize = 14.sp
        )


        Spacer(
            modifier =
                Modifier.height(7.dp)
        )


        SelectionBox(
            value = value,
            options = options,
            onSelected = onSelected,
            modifier =
                Modifier.fillMaxWidth()
        )
    }
}

File: unknownbank_flutter/lib/screens/credit_card_job_details_screen.dart
import 'package:flutter/material.dart';

import '../models/credit_card_application_data.dart';
import '../theme/app_colors.dart';

class CreditCardJobDetailsScreen extends StatefulWidget {
  final String productName;
  final CreditCardApplicationData applicationData;

  final ValueChanged<CreditCardApplicationData>
      onApplicationDataChange;

  final VoidCallback onNext;

  const CreditCardJobDetailsScreen({
    super.key,
    required this.productName,
    required this.applicationData,
    required this.onApplicationDataChange,
    required this.onNext,
  });

  @override
  State<CreditCardJobDetailsScreen> createState() =>
      _CreditCardJobDetailsScreenState();
}

class _CreditCardJobDetailsScreenState
    extends State<CreditCardJobDetailsScreen> {
  late CreditCardApplicationData data;

  @override
  void initState() {
    super.initState();
    data = widget.applicationData;
  }

  void _update(CreditCardApplicationData updatedData) {
    setState(() {
      data = updatedData;
    });

    widget.onApplicationDataChange(updatedData);
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Stack(
        children: [
          Positioned.fill(
            child: Image.asset(
              'assets/images/full_background_light.png',
              fit: BoxFit.cover,
            ),
          ),

          SafeArea(
            child: Column(
              children: [
                SizedBox(
                  height: 56,
                  child: Stack(
                    alignment: Alignment.center,
                    children: [
                      Align(
                        alignment: Alignment.centerLeft,
                        child: Padding(
                          padding:
                              const EdgeInsets.only(left: 8),
                          child: IconButton(
                            onPressed: () {
                              Navigator.of(context).pop();
                            },
                            icon: const Icon(
                              Icons.arrow_back,
                              color: AppColors.textPrimary,
                            ),
                          ),
                        ),
                      ),

                      Padding(
                        padding: const EdgeInsets.only(
                          left: 55,
                          right: 30,
                        ),
                        child: Text(
                          'Applying for ${widget.productName}',
                          maxLines: 1,
                          overflow: TextOverflow.ellipsis,
                          style: const TextStyle(
                            fontSize: 13,
                            fontWeight: FontWeight.bold,
                            color: Color(0xFF333333),
                          ),
                        ),
                      ),
                    ],
                  ),
                ),

                const Padding(
                  padding: EdgeInsets.fromLTRB(
                    20,
                    20,
                    23,
                    0,
                  ),
                  child: Row(
                    mainAxisAlignment:
                        MainAxisAlignment.spaceBetween,
                    children: [
                      Text(
                        'Tell Us About Your Job',
                        style: TextStyle(
                          fontSize: 15,
                          fontWeight: FontWeight.bold,
                          color: Color(0xFF333333),
                        ),
                      ),

                      Text(
                        '60%',
                        style: TextStyle(
                          fontSize: 14,
                          fontWeight: FontWeight.w500,
                          color: Color(0xFF0EAF5F),
                        ),
                      ),
                    ],
                  ),
                ),

                const SizedBox(height: 11),

                Padding(
                  padding: const EdgeInsets.symmetric(
                    horizontal: 20,
                  ),
                  child: ClipRRect(
                    borderRadius:
                        BorderRadius.circular(100),
                    child: const LinearProgressIndicator(
                      value: 0.60,
                      minHeight: 6,
                      backgroundColor:
                          Color(0xFFE6E8EA),
                      valueColor:
                          AlwaysStoppedAnimation(
                        Color(0xFF0EAF5F),
                      ),
                    ),
                  ),
                ),

                const SizedBox(height: 23),

                Expanded(
                  child: SingleChildScrollView(
                    padding: const EdgeInsets.fromLTRB(
                      20,
                      0,
                      20,
                      100,
                    ),
                    child: Column(
                      children: [
                        _TextField(
                          label: 'Employer Name',
                          value: data.employerName,
                          onChanged: (value) {
                            _update(
                              data.copyWith(
                                employerName: value,
                              ),
                            );
                          },
                        ),

                        const SizedBox(height: 18),

                        _DropdownField(
                          label: 'Occupation',
                          value: data.occupation,
                          options: const [
                            'Sales Manager',
                            'Manager',
                            'Executive',
                            'Engineer',
                            'Consultant',
                            'Others',
                          ],
                          onChanged: (value) {
                            _update(
                              data.copyWith(
                                occupation: value,
                              ),
                            );
                          },
                        ),

                        const SizedBox(height: 18),

                        _DropdownField(
                          label: 'Sector',
                          value: data.sector,
                          options: const [
                            'Logistic',
                            'Banking',
                            'Technology',
                            'Retail',
                            'Manufacturing',
                            'Government',
                            'Others',
                          ],
                          onChanged: (value) {
                            _update(
                              data.copyWith(
                                sector: value,
                              ),
                            );
                          },
                        ),

                        const SizedBox(height: 18),

                        _DropdownField(
                          label: 'Employment Type',
                          value: data.employmentType,
                          options: const [
                            'Private Employed',
                            'Government',
                            'Self Employed',
                          ],
                          onChanged: (value) {
                            _update(
                              data.copyWith(
                                employmentType: value,
                              ),
                            );
                          },
                        ),

                        const SizedBox(height: 18),

                        _DropdownField(
                          label:
                              'Business Classification',
                          value:
                              data.businessClassification,
                          options: const [
                            'Private Limited',
                            'Public Limited',
                            'Partnership',
                            'Sole Proprietor',
                            'Government',
                          ],
                          onChanged: (value) {
                            _update(
                              data.copyWith(
                                businessClassification:
                                    value,
                              ),
                            );
                          },
                        ),

                        const SizedBox(height: 18),

                        _DropdownField(
                          label:
                              'Length of Service (Year)',
                          value:
                              data.yearsOfService,
                          options: List.generate(
                            31,
                            (index) =>
                                index.toString(),
                          ),
                          onChanged: (value) {
                            _update(
                              data.copyWith(
                                yearsOfService: value,
                              ),
                            );
                          },
                        ),

                        const SizedBox(height: 18),

                        _DropdownField(
                          label:
                              'Length of Service (Months)',
                          value:
                              data.monthsOfService,
                          options: List.generate(
                            12,
                            (index) =>
                                index.toString(),
                          ),
                          onChanged: (value) {
                            _update(
                              data.copyWith(
                                monthsOfService: value,
                              ),
                            );
                          },
                        ),
                      ],
                    ),
                  ),
                ),
              ],
            ),
          ),

          Positioned(
            left: 16,
            right: 16,
            bottom: 16,
            child: SafeArea(
              top: false,
              child: SizedBox(
                height: 46,
                child: FilledButton(
                  onPressed: widget.onNext,
                  style: FilledButton.styleFrom(
                    backgroundColor:
                        AppColors.primary,
                    shape: RoundedRectangleBorder(
                      borderRadius:
                          BorderRadius.circular(100),
                    ),
                  ),
                  child: const Text(
                    'Next',
                    style: TextStyle(
                      fontSize: 15,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}

class _TextField extends StatefulWidget {
  final String label;
  final String value;
  final ValueChanged<String> onChanged;

  const _TextField({
    required this.label,
    required this.value,
    required this.onChanged,
  });

  @override
  State<_TextField> createState() =>
      _TextFieldState();
}

class _TextFieldState extends State<_TextField> {
  late final TextEditingController controller;

  @override
  void initState() {
    super.initState();

    controller =
        TextEditingController(text: widget.value);
  }

  @override
  void dispose() {
    controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment:
          CrossAxisAlignment.start,
      children: [
        Text(
          widget.label,
          style: const TextStyle(
            fontSize: 14,
            color: Color(0xFF333333),
          ),
        ),

        const SizedBox(height: 7),

        SizedBox(
          height: 46,
          child: TextField(
            controller: controller,
            onChanged: widget.onChanged,
            style: const TextStyle(
              fontSize: 15,
              color: Color(0xFF666666),
            ),
            decoration: InputDecoration(
              filled: true,
              fillColor:
                  const Color(0xFFF7F9FB),
              contentPadding:
                  const EdgeInsets.symmetric(
                horizontal: 19,
              ),
              border: OutlineInputBorder(
                borderRadius:
                    BorderRadius.circular(8),
                borderSide: const BorderSide(
                  color: Color(0xFFDDDDDD),
                ),
              ),
              enabledBorder:
                  OutlineInputBorder(
                borderRadius:
                    BorderRadius.circular(8),
                borderSide: const BorderSide(
                  color: Color(0xFFDDDDDD),
                ),
              ),
            ),
          ),
        ),
      ],
    );
  }
}

class _DropdownField extends StatelessWidget {
  final String label;
  final String value;
  final List<String> options;
  final ValueChanged<String> onChanged;

  const _DropdownField({
    required this.label,
    required this.value,
    required this.options,
    required this.onChanged,
  });

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment:
          CrossAxisAlignment.start,
      children: [
        Text(
          label,
          style: const TextStyle(
            fontSize: 14,
            color: Color(0xFF333333),
          ),
        ),

        const SizedBox(height: 7),

        Container(
          height: 46,
          padding: const EdgeInsets.only(
            left: 19,
            right: 10,
          ),
          decoration: BoxDecoration(
            color:
                const Color(0xFFF7F9FB),
            borderRadius:
                BorderRadius.circular(8),
            border: Border.all(
              color:
                  const Color(0xFFDDDDDD),
            ),
          ),
          child: DropdownButtonHideUnderline(
            child: DropdownButton<String>(
              value: value,
              isExpanded: true,
              icon: const Icon(
                Icons.keyboard_arrow_down,
                color: Color(0xFF777777),
              ),
              style: const TextStyle(
                color: Color(0xFF666666),
                fontSize: 15,
              ),
              items: options
                  .map(
                    (option) =>
                        DropdownMenuItem<String>(
                      value: option,
                      child: Text(option),
                    ),
                  )
                  .toList(),
              onChanged: (value) {
                if (value != null) {
                  onChanged(value);
                }
              },
            ),
          ),
        ),
      ],
    );
  }
}